### PR TITLE
Document net.minecraft.util

### PR DESCRIPTION
**Pull request by apple502j**
Big class javadoc for `Identifier` (previously only 1 line), method documentation for `Util`, and several class javadocs. Also adds `package-info`.

### CODE REVIEW

File: src/packageDocs/java/net/minecraft/util/package-info.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/*
+ * This file is free for everyone to use under the Creative Commons Zero license.
+ */
+
+/**
+ * Contains utility method classes, various enums, serialization helpers, and
+ * miscellaneous classes. {@link net.minecraft.util.Util} contains most of
+ * the utility methods.
+ */
+
```
Comment (Juuxel) [inline src/packageDocs/java/net/minecraft/util/package-info.java]:
```suggestion
```

File: mappings/net/minecraft/util/StringIdentifiable.mapping [hunk 2]
```diff
@@ -1,5 +1,12 @@
 CLASS net/minecraft/class_3542 net/minecraft/util/StringIdentifiable
+	COMMENT An interface, implemented by enums, that allows the enum to be serialized
+	COMMENT by codecs. An instance is identified using a string.
```
Comment (Juuxel) [inline mappings/net/minecraft/util/StringIdentifiable.mapping:3]: Hmm, isn't this also used for state properties?

File: mappings/net/minecraft/util/StringIdentifiable.mapping [hunk 3]
```diff
@@ -1,5 +1,12 @@
...
+	COMMENT
+	COMMENT @apiNote To make an enum serializable with codecs, implement this on the enum class,
+	COMMENT implement {@link #asString} to return a unique ID, and add a {@code static final}
+	COMMENT field that holds {@linkplain #createCodec the codec for the enum}.
 	METHOD method_15434 asString ()Ljava/lang/String;
+		COMMENT {@return the string representation of the enum, used for serialization}
```
Comment (Juuxel) [inline mappings/net/minecraft/util/StringIdentifiable.mapping]:
```suggestion
		COMMENT {@return the unique string representation of the enum, used for serialization}
```

File: mappings/net/minecraft/util/ActionResult.mapping [hunk 4]
```diff
@@ -15,12 +19,16 @@ CLASS net/minecraft/class_1269 net/minecraft/util/ActionResult
...
 	METHOD method_29236 success (Z)Lnet/minecraft/class_1269;
+		COMMENT {@return an action result indicating success}
+		COMMENT
+		COMMENT <p>This returns {@link #SUCCESS} if {@code swingHand} is {@code true}, otherwise
+		COMMENT {@link #CONSUME}.
 		ARG 0 swingHand
 	METHOD method_36360 shouldIncrementStat ()Z
-		COMMENT Returns whether action performance should increment an item's 'used'
-		COMMENT statistic.
+		COMMENT {@return whether action performance should increment an item's 'used'
```
Comment (Juuxel) [inline mappings/net/minecraft/util/ActionResult.mapping]:
```suggestion
		COMMENT {@return whether action performance should increment an item's "used"
```
I'd say double quotes are more common in javadoc in general

File: mappings/net/minecraft/util/FixedBufferInputStream.mapping [hunk 5]
```diff
@@ -1,4 +1,5 @@
 CLASS net/minecraft/class_6826 net/minecraft/util/FixedBufferInputStream
+	COMMENT A buffered input stream that uses fixed buffer array.
```
Comment (Juuxel) [inline mappings/net/minecraft/util/FixedBufferInputStream.mapping]:
```suggestion
	COMMENT A buffered input stream that uses a fixed-size buffer array.
```

File: mappings/net/minecraft/util/DyeColor.mapping [hunk 6]
```diff
@@ -15,18 +16,36 @@ CLASS net/minecraft/class_1767 net/minecraft/util/DyeColor
...
+		COMMENT {@return the color used for dyed signs as RGB integer}
+		COMMENT
+		COMMENT <p>The returned value is between {@code 0} and {@code 0xFFFFFF}.
 	METHOD method_7786 byFireworkColor (I)Lnet/minecraft/class_1767;
+		COMMENT {@return the dye color whose firework color is {@code color}, or {@code null}
+		COMMENT if there is no such color}
 		ARG 0 color
 	METHOD method_7787 getColorComponents ()[F
-		COMMENT Returns the red, blue and green components of this dye color.
+		COMMENT {@return the float array containing red, blue and green components of this dye color}
```
Comment (Juuxel) [inline mappings/net/minecraft/util/DyeColor.mapping]:
```suggestion
		COMMENT {@return the float array containing the red, green and blue components of this dye color}
```
They're in that order so this is clearer imo

File: mappings/net/minecraft/util/TimeHelper.mapping [hunk 7]
```diff
@@ -1,6 +1,11 @@
 CLASS net/minecraft/class_4802 net/minecraft/util/TimeHelper
+	COMMENT A class holding utility methods related to time and durations.
 	FIELD field_33868 SECOND_IN_NANOS J
+		COMMENT The equivalent of 1 second, in nanoseconds. Is {@code 1000000000}.
 	FIELD field_33869 MILLI_IN_NANOS J
+		COMMENT The equivalent of 1 millisecond, in nanoseconds. Is {@code 1000000}.
 	METHOD method_24505 betweenSeconds (II)Lnet/minecraft/class_6019;
+		COMMENT {@return an int provider that provides number of ticks, between {@code min * 20} and
```
Comment (Juuxel) [inline mappings/net/minecraft/util/TimeHelper.mapping]:
```suggestion
		COMMENT {@return an int provider that provides a time in ticks, between {@code min * 20} and
```

File: mappings/net/minecraft/util/TextifiedException.mapping [hunk 8]
```diff
@@ -1,8 +1,11 @@
 CLASS net/minecraft/class_7437 net/minecraft/util/TextifiedException
+	COMMENT An exception that has a user-friendly {@link Text} attached.
+	COMMENT This should not be constructed directly, and instead be subclassed by the users.
```
Comment (Juuxel) [inline mappings/net/minecraft/util/TextifiedException.mapping]: Not sure about that, Mojang seems to subclass for catching purposes at least in the case of the other subclass

File: mappings/net/minecraft/util/Util.mapping [hunk 9]
```diff
@@ -284,20 +356,64 @@ CLASS net/minecraft/class_156 net/minecraft/util/Util
 		ARG 0 futures
 			COMMENT the completable futures to combine
 	METHOD method_654 make (Ljava/lang/Object;Ljava/util/function/Consumer;)Ljava/lang/Object;
+		COMMENT {@return {@code object} initialized with {@code initializer}}
+		COMMENT
+		COMMENT <p>This is useful when initializing static fields:
+		COMMENT <pre>{@code
+		COMMENT private static final Map<String, String> MAP = Util.make(new HashMap<>(), (map) -> {
+		COMMENT     map.put("example", "hello");
+		COMMENT     return map;
```
Comment (Juuxel) [inline mappings/net/minecraft/util/Util.mapping]:
```suggestion
```
It's a consumer, not a function

File: mappings/net/minecraft/util/Identifier.mapping [hunk 10]
```diff
@@ -1,5 +1,89 @@
...
+	COMMENT /}. Uppercase letters cannot be used. {@link #isValid} can be used to check whether a
+	COMMENT string is a valid identifier. When handling externally provided identifiers, it should
+	COMMENT either validate or use {@link #tryParse} instead of the constructor. Another common
+	COMMENT mistake is using a formatted string with {@code %d} or {@code %f} to construct an
+	COMMENT identifier without specifying the locate explicitly, as they are not guaranteed to be
+	COMMENT ASCII digits in certain locales. Use {@link String#format(Locale, String, Object[])}
+	COMMENT with {@link java.util.Locale#ROOT} instead of {@link String#formatted}.
+	COMMENT
+	COMMENT <h3 id="namespace">Namespace</h3>
+	COMMENT <p><strong>Namespace</strong> of an identifier identifies the origin of the thing.
```
Comment (Juuxel) [inline mappings/net/minecraft/util/Identifier.mapping]:
```suggestion
	COMMENT <p>The <strong>namespace</strong> of an identifier identifies the origin of the thing.
```

File: mappings/net/minecraft/util/Identifier.mapping [hunk 11]
```diff
@@ -1,5 +1,89 @@
...
+	COMMENT <p>Identifiers are formatted as {@code <namespace>:<path>}. If the namespace and colon
+	COMMENT is omitted, the namespace defaults to {@value #DEFAULT_NAMESPACE}.
+	COMMENT
+	COMMENT <p><strong>The namespace and path must contain only ASCII lowercase letters ({@code
+	COMMENT [a-z]}), ASCII digits ({@code [0-9]}), or the characters {@code _}, {@code .}, and
+	COMMENT {@code -}. </strong> The path can also contain the standard path separator {@code
+	COMMENT /}. Uppercase letters cannot be used. {@link #isValid} can be used to check whether a
+	COMMENT string is a valid identifier. When handling externally provided identifiers, it should
+	COMMENT either validate or use {@link #tryParse} instead of the constructor. Another common
+	COMMENT mistake is using a formatted string with {@code %d} or {@code %f} to construct an
```
Comment (Juuxel) [inline mappings/net/minecraft/util/Identifier.mapping:17]: Do people really use `%f` when constructing IDs? :concern:

File: mappings/net/minecraft/util/Identifier.mapping [hunk 12]
```diff
@@ -1,5 +1,89 @@
...
+	COMMENT ASCII digits in certain locales. Use {@link String#format(Locale, String, Object[])}
+	COMMENT with {@link java.util.Locale#ROOT} instead of {@link String#formatted}.
+	COMMENT
+	COMMENT <h3 id="namespace">Namespace</h3>
+	COMMENT <p><strong>Namespace</strong> of an identifier identifies the origin of the thing.
+	COMMENT For example, two mods to the game could both add an item with the ID "orange";
+	COMMENT the namespace is used to differentiate the two. (The convention is to use the ID
+	COMMENT assigned to the mod as the namespace.)
+	COMMENT
+	COMMENT <p>Namespace only determines the origin, and does not determine the type; so long as
```
Comment (Juuxel) [inline mappings/net/minecraft/util/Identifier.mapping]:
```suggestion
	COMMENT <p>A namespace only determines the source of an identifier, and does not determine its purpose; so long as
```

File: mappings/net/minecraft/util/Identifier.mapping [hunk 13]
```diff
@@ -1,5 +1,89 @@
...
+	COMMENT the identifiers are used for different purposes, it can share the namespace and path.
```
Comment (Juuxel) [inline mappings/net/minecraft/util/Identifier.mapping]:
```suggestion
	COMMENT two identifiers are used for different purposes, they can share the namespace and path.
```

File: mappings/net/minecraft/util/Identifier.mapping [hunk 14]
```diff
@@ -1,5 +1,89 @@
...
+	COMMENT {@code <namespace>:<path>} format. If the colon is missing, the created identifier
+	COMMENT has the namespace {@value #DEFAULT_NAMESPACE} and the argument is used as the path.
+	COMMENT When passed an invalid value, this returns {@code null}.</li>
+	COMMENT <li>{@link #of} creates an identifier from namespace and path.
+	COMMENT When passed an invalid value, this returns {@code null}.</li>
+	COMMENT <li>{@link #fromCommandInput} reads an identifier from command input reader.
+	COMMENT When an invalid value is read, this throws {@link #COMMAND_EXCEPTION}.</li>
+	COMMENT <li>{@link Identifier.Serializer} is a serializer for Gson.</li>
+	COMMENT <li>{@link #CODEC} can be used to serialize and deserialize an identifier using
+	COMMENT DataFixerUpper,</li>
```
Comment (Juuxel) [inline mappings/net/minecraft/util/Identifier.mapping]:
```suggestion
	COMMENT DataFixerUpper.</li>
```

File: mappings/net/minecraft/util/Identifier.mapping [hunk 15]
```diff
@@ -1,5 +1,89 @@
...
+	COMMENT </ul>
+	COMMENT
+	COMMENT <h2 id="using">Using Identifier</h2>
+	COMMENT <p>Identifiers identify several objects in the game. {@link
+	COMMENT net.minecraft.util.registry.Registry} holds objects, such as blocks and items, that are
+	COMMENT identified by an identifier. Textures are also identified using an identifier; such
+	COMMENT identifier is represented as a file path with extension, such as {@code
```
Comment (Juuxel) [inline mappings/net/minecraft/util/Identifier.mapping]:
```suggestion
	COMMENT an identifier is represented as a file path with an extension, such as {@code
```

File: mappings/net/minecraft/util/Identifier.mapping [hunk 16]
```diff
@@ -1,5 +1,89 @@
...
+	COMMENT minecraft:textures/entity/pig/pig.png}.
+	COMMENT
+	COMMENT <p>The string representation of the identifier ({@code <namespace>:<path>}) can be
+	COMMENT obtained by calling {@link #toString}. This always includes the namespace. An identifier
+	COMMENT can be converted to a translation key using {@link #toTranslationKey(String)} method.
+	COMMENT
+	COMMENT <h3 id="registrykey">RegistryKey</h3>
+	COMMENT <p>Identifier is not type-aware; {@code minecraft:tnt} could refer to a TNT block, a TNT
+	COMMENT item, or a TNT entity. To identify a registered object uniquely, {@link
+	COMMENT net.minecraft.util.registry.RegistryKey} should be used. A registry key is a combination
```
Comment (Juuxel) [inline mappings/net/minecraft/util/Identifier.mapping]:
```suggestion
	COMMENT net.minecraft.util.registry.RegistryKey} can be used. A registry key is a combination
```

**Review by Juuxel — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/util/StringIdentifiable.mapping (hunk 2 above)
Comment (apple502j) [reply mappings/net/minecraft/util/StringIdentifiable.mapping:3]: They are technically enums.

File: mappings/net/minecraft/util/TextifiedException.mapping (hunk 8 above)
Comment (apple502j) [reply mappings/net/minecraft/util/TextifiedException.mapping]:
```suggestion
```

File: mappings/net/minecraft/util/DyeColor.mapping [hunk 17]
```diff
@@ -15,18 +16,36 @@ CLASS net/minecraft/class_1767 net/minecraft/util/DyeColor
...
+		COMMENT
+		COMMENT @apiNote If out-of-range IDs are passed, this returns {@link #WHITE}.
 		ARG 0 id
 	METHOD method_7792 getName ()Ljava/lang/String;
+		COMMENT {@return the name of the dye color}
 	METHOD method_7793 byName (Ljava/lang/String;Lnet/minecraft/class_1767;)Lnet/minecraft/class_1767;
+		COMMENT {@return the dye color whose name is {@code name}, or {@code defaultColor} if
+		COMMENT there is no such color}
+		COMMENT
+		COMMENT @apiNote This returns {@code null} if and only if {@code defaultColor} is {@code null}.
```
Comment (enbrain) [inline mappings/net/minecraft/util/DyeColor.mapping]:
```suggestion
		COMMENT @apiNote This returns {@code null} only if {@code defaultColor} is {@code null}.
```

`byName("white", null)` returns a non-null value even though `defaultColor` is `null`.

File: mappings/net/minecraft/util/StringIdentifiable.mapping (hunk 2 above)
Comment (Juuxel) [reply mappings/net/minecraft/util/StringIdentifiable.mapping:3]: I was talking about the codec part. AFAIK properties, at least partially, do their own serialisation.

File: mappings/net/minecraft/util/Util.mapping [hunk 18]
```diff
@@ -1,46 +1,71 @@
...
 		ARG 0 thread
 		ARG 1 t
 	METHOD method_18349 getMainWorkerExecutor ()Ljava/util/concurrent/ExecutorService;
+		COMMENT {@return the main worker executor for miscellaneous asynchronous tasks}
 	METHOD method_18350 shutdownExecutors ()V
 	METHOD method_18839 debugRunnable (Ljava/lang/Runnable;Ljava/util/function/Supplier;)Ljava/lang/Runnable;
 		ARG 0 runnable
 		ARG 1 messageSupplier
 	METHOD method_19483 completeExceptionally (Ljava/lang/Throwable;)Ljava/util/concurrent/CompletableFuture;
+		COMMENT {@return a future that completes exceptionally with {@code throwable}}
```
Comment (enbrain) [inline mappings/net/minecraft/util/Util.mapping]:
```suggestion
		COMMENT {@return a future that is already completed exceptionally with {@code throwable}}
```

File: mappings/net/minecraft/util/Util.mapping [hunk 19]
```diff
@@ -55,16 +80,26 @@ CLASS net/minecraft/class_156 net/minecraft/util/Util
...
 	METHOD method_28124 getBootstrapExecutor ()Ljava/util/concurrent/ExecutorService;
+		COMMENT {@return the executor for asynchronous bootstrapping}
+		COMMENT
+		COMMENT <p>This is used by DataFixerUpper to build schemas.
 	METHOD method_29187 getChoiceType (Lcom/mojang/datafixers/DSL$TypeReference;Ljava/lang/String;)Lcom/mojang/datafixers/types/Type;
 		ARG 0 typeReference
 		ARG 1 id
 	METHOD method_29188 addPrefix (Ljava/lang/String;Ljava/util/function/Consumer;)Ljava/util/function/Consumer;
+		COMMENT {@return a consumer that passes the received string to {@code consumer}, with
+		COMMENT {@code prefix}}
```
Comment (enbrain) [inline mappings/net/minecraft/util/Util.mapping]:
```suggestion
		COMMENT {@return a consumer that first prepends {@code prefix} to its input
		COMMENT string and passes the result to {@code consumer}}
```

File: mappings/net/minecraft/util/Util.mapping [hunk 20]
```diff
@@ -259,17 +312,36 @@ CLASS net/minecraft/class_156 net/minecraft/util/Util
 			COMMENT a function that, when given the size of {@code values},
 			COMMENT returns a map for storing the indices of the values
 	METHOD method_44893 getFormattedCurrentTime ()Ljava/lang/String;
+		COMMENT {@return the current time formatted using {@link #DATE_TIME_FORMATTER}}
 	METHOD method_645 previous (Ljava/lang/Iterable;Ljava/lang/Object;)Ljava/lang/Object;
+		COMMENT {@return the item preceding {@code object} in {@code iterable}}
+		COMMENT
+		COMMENT <p>If {@code object} is not in {@code iterable}, this returns the last item of the iterable.
```
Comment (enbrain) [inline mappings/net/minecraft/util/Util.mapping:319]: Might be worth mentioning how the equality is checked (`equals` or `==`).

File: mappings/net/minecraft/util/Util.mapping [hunk 21]
```diff
@@ -284,20 +356,63 @@ CLASS net/minecraft/class_156 net/minecraft/util/Util
...
+		COMMENT <p>This should be used to display or store the current time. {@link #getMeasuringTimeMs}
+		COMMENT should be used for determining the duration between two calls.
+		COMMENT
+		COMMENT @see #getMeasuringTimeMs
+		COMMENT @see #getMeasuringTimeNano
 	METHOD method_660 next (Ljava/lang/Iterable;Ljava/lang/Object;)Ljava/lang/Object;
+		COMMENT {@return the item succeeding {@code object} in {@code iterable}}
+		COMMENT
+		COMMENT <p>If {@code object} is {@code null} or is not in {@code iterable}, this
+		COMMENT returns the first item of the iterable.
```
Comment (enbrain) [inline mappings/net/minecraft/util/Util.mapping]: Same as above.

File: mappings/net/minecraft/util/Util.mapping [hunk 22]
```diff
@@ -319,17 +434,39 @@ CLASS net/minecraft/class_156 net/minecraft/util/Util
 		METHOD hashCode (Ljava/lang/Object;)I
 			ARG 1 o
 	CLASS class_158 OperatingSystem
+		COMMENT An enum representing the operating system of the current platform.
+		COMMENT This defines the behavior for opening links.
+		COMMENT An instance can be obtained via {@link Util#getOperatingSystem}.
```
Comment (enbrain) [inline mappings/net/minecraft/util/Util.mapping]:
```suggestion
		COMMENT The current one can be obtained via {@link Util#getOperatingSystem}.
```

File: mappings/net/minecraft/util/Identifier.mapping [hunk 23]
```diff
@@ -1,5 +1,89 @@
 CLASS net/minecraft/class_2960 net/minecraft/util/Identifier
-	COMMENT The namespace and path must contain only lowercase letters ([a-z]), digits ([0-9]), or the characters '_', '.', and '-'. The path can also contain the standard path separator '/'.
+	COMMENT An identifier used to identify things. This is also known as "resource location",
+	COMMENT "namespaced ID", "location", or just "ID". This is a non-typed immutable object,
+	COMMENT and identifies things using a combination of namespace and path. Identifiers should
+	COMMENT always be compared using {@link #equals} method, not {@code ==}.
+	COMMENT
+	COMMENT <h2 id="format">Format</h2>
+	COMMENT <p>Identifiers are formatted as {@code <namespace>:<path>}. If the namespace and colon
+	COMMENT is omitted, the namespace defaults to {@value #DEFAULT_NAMESPACE}.
```
Comment (enbrain) [inline mappings/net/minecraft/util/Identifier.mapping]:
```suggestion
	COMMENT are omitted, the namespace defaults to {@value #DEFAULT_NAMESPACE}.
```

File: mappings/net/minecraft/util/StringIdentifiable.mapping (hunk 2 above)
Comment (apple502j) [reply mappings/net/minecraft/util/StringIdentifiable.mapping:3]: Yeah; but I think this is acceptable.

File: mappings/net/minecraft/util/Util.mapping (hunk 21 above)
Comment (enbrain) [inline mappings/net/minecraft/util/Util.mapping]: If `object` is not in `iterable`, the while loop never terminates.

image

File: mappings/net/minecraft/util/Util.mapping [hunk 24]
```diff
@@ -259,17 +312,37 @@ CLASS net/minecraft/class_156 net/minecraft/util/Util
 			COMMENT a function that, when given the size of {@code values},
 			COMMENT returns a map for storing the indices of the values
 	METHOD method_44893 getFormattedCurrentTime ()Ljava/lang/String;
+		COMMENT {@return the current time formatted using {@link #DATE_TIME_FORMATTER}}
 	METHOD method_645 previous (Ljava/lang/Iterable;Ljava/lang/Object;)Ljava/lang/Object;
+		COMMENT {@return the item preceding {@code object} in {@code iterable}}
+		COMMENT
+		COMMENT @implNote If {@code object} is not in {@code iterable}, this enters into an infinite loop.
```
Comment (enbrain) [inline mappings/net/minecraft/util/Util.mapping]: Unlike `next`, `previous` does not enter into an infinite loop.

I think these methods *expect* `object` to be in `iterable`. (If that's the case, these methods should throw `IllegalArgumentException` though.)

File: mappings/net/minecraft/util/Identifier.mapping [hunk 25]
```diff
@@ -1,5 +1,89 @@
...
+	COMMENT the namespace is used to differentiate the two. (The convention is to use the ID
+	COMMENT assigned to the mod as the namespace.)
+	COMMENT
+	COMMENT <p>A namespace only determines the source of an identifier, and does not determine its purpose; so long as
+	COMMENT two identifiers are used for different purposes, they can share the namespace and path.
+	COMMENT For example, the identifier {@code minecraft:dirt} is shared by blocks and items.
+	COMMENT There is no need to change the identifier to, say, {@code minecraft_block:dirt} or
+	COMMENT {@code minecraft_item:dirt}.
+	COMMENT
+	COMMENT <p>Several namespaces are reserved for vanilla use. While those identifier can be used for
```
Comment (enbrain) [inline mappings/net/minecraft/util/Identifier.mapping]:
```suggestion
	COMMENT <p>Several namespaces are reserved for vanilla use. While those identifiers can be used for
```

File: mappings/net/minecraft/util/Util.mapping [hunk 26]
```diff
@@ -319,17 +436,39 @@ CLASS net/minecraft/class_156 net/minecraft/util/Util
...
 	CLASS class_158 OperatingSystem
+		COMMENT An enum representing the operating system of the current platform.
+		COMMENT This defines the behavior for opening links.
+		COMMENT The current one can be obtained via {@link Util#getOperatingSystem}.
 		FIELD field_34894 name Ljava/lang/String;
 		METHOD <init> (Ljava/lang/String;ILjava/lang/String;)V
 			ARG 3 name
 		METHOD method_38649 getName ()Ljava/lang/String;
 		METHOD method_669 open (Ljava/net/URL;)V
+			COMMENT Opens {@code url}. If this points to a HTTP(S) URL, it is usually opened using
```
Comment (enbrain) [inline mappings/net/minecraft/util/Util.mapping]:
```suggestion
			COMMENT Opens {@code url}. If this points to an HTTP(S) URL, it is usually opened using
```

File: mappings/net/minecraft/util/Util.mapping [hunk 27]
```diff
@@ -319,17 +436,39 @@ CLASS net/minecraft/class_156 net/minecraft/util/Util
...
+			COMMENT the system's default browser. Otherwise, it is opened directly.
+			COMMENT
+			COMMENT <p><strong>Always validate the passed URL's schema</strong> as some values can
+			COMMENT execute code.
 			ARG 1 url
 		METHOD method_670 open (Ljava/lang/String;)V
+			COMMENT Opens {@code uri}. If this points to a HTTP(S) URI, it is usually opened using
```
Comment (enbrain) [inline mappings/net/minecraft/util/Util.mapping]:
```suggestion
			COMMENT Opens {@code uri}. If this points to an HTTP(S) URI, it is usually opened using
```

File: mappings/net/minecraft/util/Util.mapping [hunk 28]
```diff
@@ -319,17 +436,39 @@ CLASS net/minecraft/class_156 net/minecraft/util/Util
...
+			COMMENT execute code.
 			ARG 1 uri
 		METHOD method_672 open (Ljava/io/File;)V
+			COMMENT Opens {@code file}.
+			COMMENT
+			COMMENT <p><strong>Do not pass untrusted file to this method</strong> as some values can
+			COMMENT execute code.
 			ARG 1 file
 		METHOD method_673 open (Ljava/net/URI;)V
+			COMMENT Opens {@code uri}. If this points to a HTTP(S) URI, it is usually opened using
```
Comment (enbrain) [inline mappings/net/minecraft/util/Util.mapping]:
```suggestion
			COMMENT Opens {@code uri}. If this points to an HTTP(S) URI, it is usually opened using
```

**Review by Juuxel — APPROVED** [review]

**Review by enbrain — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**